iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/connection/PipeEventCollector.java

```diff
@@ -132,14 +127,20 @@ private void parseAndCollectEvent(final PipeTsFileInsertionEvent sourceEvent) th
 
     try {
       for (final TabletInsertionEvent parsedEvent : sourceEvent.toTabletInsertionEvents()) {
-        hasNoGeneratedEvent = false;
-        collectEvent(parsedEvent);
+        collectParsedRawTableEvent((PipeRawTabletInsertionEvent) parsedEvent);
       }
     } finally {
       sourceEvent.close();
     }
   }
 
+  private void collectParsedRawTableEvent(final PipeRawTabletInsertionEvent parsedEvent) {
+    if (!parsedEvent.hasNoNeedParsingAndIsEmpty()) {
+      hasNoGeneratedEvent = false;
+      collectEvent(parsedEvent);
```

Comment:
Is there anything to do with the receiver? I think it's better to add some defending logic to prevent NPE...